src/ServiceInsight/Shell/ServiceControlConnectionViewModel.cs

```diff
@@ -13,12 +14,26 @@
 
     public class ServiceControlConnectionViewModel : Screen
     {
-        public const string ConnectingToServiceControl = "Connecting to ServiceControl...";
+        const string ConnectingToServiceControl = "Connecting to ServiceControl...";
+        const string ConnectionErrorMessage = "There was an error connecting to ServiceControl. Either the address is not valid or the service is down.";
+        const string CertValidationErrorMessage = "There was an error connecting to ServiceControl. SSL certificate is not valid.";
+        static bool certValidationFailed;
 
         ISettingsProvider settingsProvider;
         ProfilerSettings appSettings;
         IContainer container;
 
+        static ServiceControlConnectionViewModel()
+        {
+            ServicePointManager.ServerCertificateValidationCallback = (sender, certificate, chain, errors) => OnCertificateValidationFailed();
```

Comment:
This is a static delegate, but we assign for this `ServiceControlConnectionViewModel` only. This doesn't look right to me. I followed `IsValidUrl` and as far as I understand, eventually, it calls `DefaultServiceControl`, which underneath issues a rest request against the specified `url`. I know that passing the delegate down the drain, till it meets the `IRestClient` implementation might be hard, but we'd ensure that we don't loose anything (or override something for future versions).

Reply:
Problem is that the callback is static. If you look more, DefaultServiceControl class has also subscribed to the same delegate. Now we have some options:

- Move the delegate to the ViewModel class? It doesn't sound right as it is not _just_ a presentation concern.

- Have two delegates attached to the callback? This is what we have. The callback uses the _last_ returned result, so since we don't control which one is the last one, I had to use the exact same code (e.g. return the same flag) to ensure it follows the same logic.

- Have another static callback in DefaultServiceControl which gets called as a part of ServerCertificateValidationCallback. That helps with duplicate `return` logic, but still not great.

I'll update the PR with the third option, see what you think.